llvm/lib/Target/RISCV/RISCVInstrInfoXAndes.td

```diff
@@ -53,6 +53,42 @@ def simm20_lsb000 : Operand<XLenVT> {
   let DecoderMethod = "decodeSImmOperandAndLslN<20, 3>";
 }
 
+// Predicate: True if immediate is a power of 2.
+def PowerOf2 : PatLeaf<(imm), [{
+  if (N->getValueType(0) == MVT::i32)
+    return isPowerOf2_32(N->getZExtValue());
+  else if (N->getValueType(0) == MVT::i64)
+    return isPowerOf2_64(N->getZExtValue());
```

Comment:
It should be ok to use isPowerOf2_64 for both i32 and i64. The getZExtValue will pad i32 to 64-bits with 0s which won't change whether it is a power of 2.